go/test/endtoend/multipooler/rpc_manager_replication_api_test.go

```diff
@@ -505,13 +514,323 @@ func TestReplicationAPIs(t *testing.T) {
 		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
 		defer cancel()
 
-		stopReq := &multipoolermanagerdatapb.StopReplicationRequest{}
+		stopReq := &multipoolermanagerdatapb.StopReplicationRequest{
+			Mode: multipoolermanagerdatapb.ReplicationPauseMode_REPLICATION_PAUSE_MODE_REPLAY_ONLY,
+			Wait: true,
+		}
 		_, err = primaryManagerClient.StopReplication(ctx, stopReq)
 		require.Error(t, err, "StopReplication should fail on primary")
 		assert.Contains(t, err.Error(), "operation not allowed", "Error should indicate operation not allowed on PRIMARY")
 		t.Log("Confirmed: StopReplication correctly rejected on PRIMARY pooler")
 	})
 
+	t.Run("StopReplication_ReceiverOnly_Wait", func(t *testing.T) {
+		// This test verifies that RECEIVER_ONLY mode with wait=true:
+		// 1. Clears primary_conninfo and disconnects the WAL receiver
+		// 2. Does NOT pause WAL replay (replay continues)
+		// 3. Waits for receiver to fully disconnect before returning
+
+		setupPoolerTest(t, setup, WithDropTables("test_receiver_only"))
```

Comment:
I don't think the WithDropTables() is needed here? Doesn't seem like the test creates that table.